src/trossen_rslidar_node.cpp

```diff
@@ -0,0 +1,283 @@
+/*********************************************************************************************************************
+Copyright (c) 2020 RoboSense
+All rights reserved
+
+Modified by: Trossen Robotics
+Copyright (c) 2024 Trossen Robotics
+
+By downloading, copying, installing or using the software you agree to this license. If you do not agree to this
+license, do not download, install, copy or use the software.
+
+License Agreement
+For RoboSense LiDAR SDK Library
+(3-clause BSD License)
+
+Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following
+disclaimer in the documentation and/or other materials provided with the distribution.
+
+3. Neither the names of the RoboSense, nor Suteng Innovation Technology, nor the names of other contributors may be used
+to endorse or promote products derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************************************************************/
+
+
+#include <trossen_rslidar_node.hpp>
+
+namespace robosense
+{
+namespace lidar
+{
+
+PointCloudLFNode::PointCloudLFNode(const rclcpp::NodeOptions & options)
+: nav2_util::LifecycleNode("trossen_rslidar", "", options)
+{
+    declare_parameter<std::string>("ros_frame_id", "rslidar");
+    declare_parameter<std::string>("ros_send_point_cloud_topic", "points");
+    declare_parameter<bool>("ros_send_by_rows", false);
+    declare_parameter<bool>("show_driver_config", false);
+
+    // input related
+    declare_parameter<uint16_t>("msop_port", 6699);
+    declare_parameter<uint16_t>("difop_port", 7788);
+    declare_parameter<std::string>("host_address", "0.0.0.0");
+    declare_parameter<std::string>("group_address", "0.0.0.0");
+    declare_parameter<bool>("use_vlan", false);
+    declare_parameter<std::string>("pcap_path", "");
+    declare_parameter<float>("pcap_rate", 1);
```

Comment:
For float params, let's have the defaults be explicitly floats i.e. `1.0` instead of `1`. Let's change that everywhere.
```suggestion
    declare_parameter<float>("pcap_rate", 1.0);
```